react-js/react-part-2.md

```diff
@@ -0,0 +1,219 @@
+# React Part 2
+
+## Projected Time
+
+2 hours 30 mins
+
+- Lesson: 1 hour
+- Guided Practice: 30 mins
+- Independent Practice: 1 hour
+
+### Prerequisites
+
+- [React](react.md)
+
+### Motivation
+
+In introduction to [React](/react.md) we discussed about how React makes websites interactive and reduces jQuery code. Now the problem arises if we want to store some information which intent to change and due to change of that information the UI changes i.e the page is re-rendered. Thus, component state is used for such cases.
+
+### Objectives
+
+- Component state
+- State vs Props
+- State are Immutable
+
+### Lesson
+
+- [Introduction to component state in React (video)](https://www.youtube.com/watch?v=e5n9j9n83OM)
+- [Component state in React slides](https://docs.google.com/presentation/d/1Trj_7vWYe79_mDboDxN9rdT94yjY5Dt55xtU1k3dUsw/edit?usp=sharing)
+
+### Common Mistakes & Misconceptions
+
+1. The scope of the state is local to the component.
+2. Don't forget to update state immutably.
+
+### Guided Practice
+
+- Define state.
+  ```
```

Comment:
All code snippets need to be marked as JavaScript so they will have proper highlighting.

This was an earlier unresolved comment. This is covered under "Syntax Highlighting" on https://guides.github.com/features/mastering-markdown/